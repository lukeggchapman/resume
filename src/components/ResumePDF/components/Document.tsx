import React from 'react'
import {
  Page,
  Text,
  Link as PDFLink,
  View,
  Document,
  StyleSheet,
} from '@react-pdf/renderer'

import typography, { FAIcon } from './typography'
import Section from './common/Section'
import Experience from './Experience'
import Education from './Education'
import Skills from './Skills'
import ResumeData from '../ResumeData'

export interface DocumentProps {
  data: ResumeData
}

const stylesObj = {
  page: {
    flexDirection: 'column',
    backgroundColor: 'white',
    fontFamily: 'Crimson Text',
    fontSize: 12,
    color: '#333',
    padding: '20 30',
  },
  header: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginBottom: 10,
  },
  info: {
    alignItems: 'right',
    fontFamily: 'PT Sans',
  },
  body: {
    flexDirection: 'row',
    flexGrow: 1,
  },
  content: {
    flex: 6,
  },
  sidePanel: {
    flex: 1,
    marginLeft: 10,
  },
  footer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
}
const styles = StyleSheet.create(stylesObj)

const ResumeDocument = ({ data }: DocumentProps) => {
  return (
    <Document title="Luke Chapman Résumé">
      <Page size="A4" style={styles.page}>
        <View style={styles.header}>
          <Text style={typography.h1}>Luke Chapman</Text>
          <View style={styles.info}>
            <Text>
              <FAIcon icon="envelope" fas />
              {` `}
              <PDFLink src="mailto:lukeggchapman@gmail.com">
                LukeGGChapman@gmail.com
              </PDFLink>
            </Text>
            <Text>
              <FAIcon icon="phone" fas />
              {` `}
              <PDFLink src="tel:+61401516361">+61 401 516 361</PDFLink>
            </Text>
          </View>
        </View>
        <View style={styles.body}>
          <View style={styles.content}>
            <Section title="Experience">
              <Experience data={data.experience} />
            </Section>
            <Section title="Education">
              <Education data={data.education} />
            </Section>
          </View>
          <View style={styles.sidePanel}>
            <Section title="Skills">
              <Skills data={data.skills} />
            </Section>
          </View>
        </View>
        <View style={[typography.h4, styles.footer]}>
          <Text>
            <FAIcon icon="github" fab />
            {` `}
            lukeggchapman
          </Text>
          <Text>
            <FAIcon icon="linkedin" fab />
            {` `}
            lukeggchapman
          </Text>
          <Text>
            <FAIcon icon="react" fab />
            {` `}
            Created with React.
            {' Sourcecode '}
            <PDFLink src="https://bit.ly/2VgE4pH">
              https://bit.ly/2VgE4pH
            </PDFLink>
            {` `}
            <PDFLink src="" />
          </Text>
        </View>
      </Page>
    </Document>
  )
}

export default ResumeDocument
