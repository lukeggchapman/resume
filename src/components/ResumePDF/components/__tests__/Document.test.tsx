import React from 'react'
import { shallow } from 'enzyme'

import Document, { DocumentProps } from '../Document'

jest.mock('../Experience', () => 'Experience')
jest.mock('../Education', () => 'Education')
jest.mock('../Skills', () => 'Skills')

const baseProps: DocumentProps = {
  data: {
    experience: ('experienceTest' as unknown) as DocumentProps['data']['experience'],
    skills: ('skillsTest' as unknown) as DocumentProps['data']['skills'],
  },
}

describe('Document', () => {
  const render = (props?: Partial<DocumentProps>) =>
    shallow(<Document {...{ ...baseProps, ...props }} />)

  it('renders correctly', () => {
    expect(render()).toMatchSnapshot()
  })
})
