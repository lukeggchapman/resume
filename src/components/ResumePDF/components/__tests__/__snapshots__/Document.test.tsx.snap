// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Document renders correctly 1`] = `
<Document
  title="Luke Chapman Résumé"
>
  <Page
    size="A4"
    style={
      Array [
        "body",
        Object {
          "backgroundColor": "white",
          "flexDirection": "column",
          "padding": "20 30",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 10,
        }
      }
    >
      <Text
        style="h1"
      >
        Luke Chapman
      </Text>
      <View
        style={
          Object {
            "alignItems": "right",
            "fontFamily": "PT Sans",
          }
        }
      >
        <Text>
          <FAIcon
            fas={true}
            icon="envelope"
          />
           
          <Link
            src="mailto:lukeggchapman@gmail.com"
          >
            LukeGGChapman@gmail.com
          </Link>
        </Text>
        <Text>
          <FAIcon
            fas={true}
            icon="phone"
          />
           
          <Link
            src="tel:+61401516361"
          >
            +61 401 516 361
          </Link>
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "flexGrow": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 6,
          }
        }
      >
        <Section
          title="Experience"
        >
          <Experience
            data="experienceTest"
          />
        </Section>
        <Section
          title="Education"
        >
          <Education
            data="educationTest"
          />
        </Section>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "marginLeft": 10,
          }
        }
      >
        <Section
          title="Skills"
        >
          <Skills
            data="skillsTest"
          />
        </Section>
      </View>
    </View>
    <View
      style={
        Array [
          "h4",
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
        ]
      }
    >
      <Text>
        <FAIcon
          fab={true}
          icon="github"
        />
         
        <Link
          src="https://github.com/lukeggchapman"
        >
          lukeggchapman
        </Link>
      </Text>
      <Text>
        <FAIcon
          fab={true}
          icon="linkedin"
        />
         
        <Link
          src="https://www.linkedin.com/in/lukeggchapman"
        >
          lukeggchapman
        </Link>
      </Text>
      <Text>
        <FAIcon
          fab={true}
          icon="react"
        />
         
        Created with React.
         Source code 
        <Link
          src="https://bit.ly/2VgE4pH"
        >
          https://bit.ly/2VgE4pH
        </Link>
      </Text>
    </View>
  </Page>
</Document>
`;
